packages/client/src/encryption/LitProtocolFacade.ts

```diff
@@ -0,0 +1,155 @@
+import { StreamPermission, streamPermissionToSolidityType } from '../permission'
+import { ConfigInjectionToken, StrictStreamrClientConfig } from '../Config'
+import { GroupKey } from './GroupKey'
+import { Logger, withRateLimit } from '@streamr/utils'
+import { LoggerFactory } from '../utils/LoggerFactory'
+
+const logger = new Logger(module)
+
+const chain = 'polygon'
+
+const LIT_PROTOCOL_CONNECT_INTERVAL = 60 * 60 * 1000 // 1h
+
+const formEvmContractConditions = (streamRegistryChainAddress: string, streamId: StreamID) => ([
+    {
+        contractAddress: streamRegistryChainAddress,
+        chain,
+        functionName: 'hasPermission',
+        functionParams: [streamId, ':userAddress', `${streamPermissionToSolidityType(StreamPermission.SUBSCRIBE)}`],
+        functionAbi: {
+            inputs: [
+                {
+                    name: "streamId",
+                    type: "string"
+                },
+                {
+                    name: "user",
+                    type: "address"
+                },
+                {
+                    name: "permissionType",
+                    type: "uint8"
+                }
+            ],
+            name: "hasPermission",
+            outputs: [
+                {
+                    name: "userHasPermission",
+                    type: "bool"
+                }
+            ],
+            stateMutability: "view",
+            type: "function"
+        },
+        returnValueTest: {
+            key: "userHasPermission",
+            comparator: '=',
+            value: "true",
+        },
+    }
+])
+
+const signAuthMessage = async (authentication: Authentication) => {
+    const domain = "dummy.com"
+    const uri = "https://dummy.com"
+    const statement = "dummy"
+    const addressInChecksumCase = ethers.utils.getAddress(await authentication.getAddress())
+    const siweMessage = new siwe.SiweMessage({
+        domain,
+        uri,
+        statement,
+        address: addressInChecksumCase,
+        version: "1",
+        chainId: 1
+    })
+    const messageToSign = siweMessage.prepareMessage()
+    const signature = await authentication.createMessageSignature(messageToSign)
+    return {
+        sig: signature,
+        derivedVia: "web3.eth.personal.sign",
+        signedMessage: messageToSign,
+        address: addressInChecksumCase
+    }
+}
+
+/**
+ * This class only operates with Polygon production network and therefore ignores contracts config.
+ */
+@scoped(Lifecycle.ContainerScoped)
+export class LitProtocolFacade {
+    private readonly logger: Logger
+    private litNodeClient?: LitJsSdk.LitNodeClient
+    private connectLitNodeClient?: () => Promise<void>
+
+    constructor(
+        @inject(AuthenticationInjectionToken) private readonly authentication: Authentication,
+        @inject(ConfigInjectionToken) private readonly config: Pick<StrictStreamrClientConfig, 'contracts' | 'encryption'>,
+        private readonly loggerFactory: LoggerFactory
```

Comment:
The `loggerFactory` is not needed a field of `LitProtocolFacade`:

```suggestion
        loggerFactory: LoggerFactory
```

Could also consider defining all fields their types explicitly, i.e. not using the `private` modifier shortcut?  That's the "convention" we are currently using, and it makes it easier to see the complete list of class fields in one place.

The list of fields is usually important first place to check when someone is learning the functionality of a new class. Therefore having the list increases the code readability significantly, which is maybe more important compared to the benefits of the `private` modifier.